src/libs/OptionsListUtils.js

```diff
@@ -537,12 +537,14 @@ function createOption(accountIDs, personalDetails, report, reportActions = {}, {
  * @returns {Boolean}
  */
 function isSearchStringMatch(searchValue, searchText, participantNames = new Set(), isChatRoom = false) {
-    const searchWords = _.compact(uniqFast([searchValue, ..._.map(searchValue.replace(/,/g, ' ').split(' '), (word) => word.trim())]));
+    const searchWords = new Set(searchValue.replace(/,/g, ' ').split(' '));
     const valueToSearch = searchText && searchText.replace(new RegExp(/&nbsp;/g), '');
-    return _.some(searchWords, (word) => {
+    let matching = true;
+    searchWords.forEach((word) => {
         const matchRegex = new RegExp(Str.escapeForRegExp(word), 'i');
-        return matchRegex.test(valueToSearch) || (!isChatRoom && participantNames.has(word));
+        matching &&= matchRegex.test(valueToSearch) || (!isChatRoom && participantNames.has(word));
```

Comment:
I think `&&=` here is clever code but makes understanding more difficult, in a function that is already quite complex

Couldn't we switch for this?
```
    searchWords.forEach((word) => {
        // returning early if !matching, and avoids calculating regex below
        if(!matching) {
            return;
        }
    
        const matchRegex = new RegExp(Str.escapeForRegExp(word), 'i');
        
        // if matching is true, assign the new value
        matching = matchRegex.test(valueToSearch) || (!isChatRoom && participantNames.has(word));
    });
    return matching;
```

Reply:
@thiagobrez I was going to suggest this as well.